float32 prop_demand
